sdk/azidentity/aad_identity_client.go

```diff
@@ -81,7 +113,7 @@ func (c *aadIdentityClient) authenticate(ctx context.Context, tenantID string, c
 		return c.createAccessToken(resp)
 	}
 
-	return nil, &AuthenticationFailedError{inner: newAuthenticationResponseError(resp)}
+	return nil, &AuthenticationFailedError{inner: newAADAuthenticationFailedError(resp)}
```

Comment:
Avoid wrapping.

Reply:
Leaving wrapping for now. If you want I can open an issue to discuss this.